synapse/http/client.py

```diff
@@ -819,6 +820,128 @@ def __init__(
             )
 
 
+class SimpleReplicationClient(BaseHttpClient):
+    """No frills client for connecting to Replication endpoints.
+
+    Uses existing BaseHttpClient methods but replaces the 'agent' used to make the
+    request with one that supports HTTP and HTTPS.
+    Attributes:
+            endpoints.
+        agent: The custom Twisted Agent used for constructing the connection.
+    """
+
+    def __init__(
+        self,
+        hs: "HomeServer",
+    ):
+        """
+        Args:
+            hs
+        """
+        super().__init__(hs)
+
+        # Use a pool, but a very small one.
+        pool = HTTPConnectionPool(self.reactor)
+        pool.maxPersistentPerHost = 5
+        pool.cachedConnectionTimeout = 2 * 60
+
+        self.agent: IAgent = ReplicationAgent(
+            hs.get_reactor(),
+            hs,
+            contextFactory=hs.get_http_client_context_factory(),
+            pool=pool,
+        )
+
+    async def request(
+        self,
+        method: str,
+        uri: str,
+        data: Optional[bytes] = None,
+        headers: Optional[Headers] = None,
+    ) -> IResponse:
+        """
+        Args:
+            method: HTTP method to use.
+            uri: URI to query.
+            data: Data to send in the request body, if applicable.
+            headers: Request headers.
+
+        Returns:
+            Response object, once the headers have been read.
+
+        Raises:
+            RequestTimedOutError if the request times out before the headers are read
+
+        """
+        outgoing_requests_counter.labels(method).inc()
```

Comment:
This was duplicated from the base class. Should it be it's own counter, considering there is are counters in [`replication/http/_base.py`](https://github.com/matrix-org/synapse/blob/197fbb123bc1a08a3d64e3acd658d3765f86d2fc/synapse/replication/http/_base.py#L44) as well? Or just remove it. *Technically*, it is an outgoing request even if it's not going external to the server(broadly speaking).

Reply:
I thought I had replied to this, sorry.

I think this is fine to leave for now. We might want to separate it later, but we aren't double counting, which is the most important thing.